fdb-record-layer-spatial/src/main/java/com/geophile/z/async/SpatialJoinAsync.java

```diff
@@ -0,0 +1,202 @@
+{
+    /**
+     * Creates and configures a new SpatialJoin object.
+     *
+     * @param duplicates    Indicates whether the spatial join will suppress duplicates.
+     * @param filter        Used to eliminate false positives from the spatial join output.
+     * @param leftObserver  Used to monitor operations on the left input to the spatial join.
+     * @param rightObserver Used to monitor operations on the right input to the spatial join.
+     * @param <LEFT>        Type of object passed to the left filter argument.
+     * @param <RIGHT>       TYpe of object passed to the right filter argument.
+     * @return A configured SpatialJoin object. All spatial joins computed using it will use the configuration
+     * specified by the above arguments.
+     */
+    public static <LEFT, RIGHT> SpatialJoinAsync<LEFT, RIGHT> newSpatialJoin(Duplicates duplicates,
+                                                                             Filter<LEFT, RIGHT> filter,
+                                                                             InputObserver leftObserver,
+                                                                             InputObserver rightObserver)
+    {
+        return new SpatialJoinImplAsync<>(duplicates, filter, leftObserver, rightObserver);
+    }
+
+    /**
+     * Creates and configures a new SpatialJoin object.
+     *
+     * @param duplicates Indicates whether the spatial join will suppress duplicates.
+     * @param filter     Used to eliminate false positives from the spatial join output.
+     * @param <LEFT>     Type of object passed to the left filter argument.
+     * @param <RIGHT>    TYpe of object passed to the right filter argument.
+     * @return A configured SpatialJoin object. All spatial joins computed using it will use the configuration
+     * specified by the above arguments.
+     */
+    public static <LEFT, RIGHT> SpatialJoinAsync<LEFT, RIGHT> newSpatialJoin(Duplicates duplicates,
+                                                                             Filter<LEFT, RIGHT> filter)
+    {
+        return new SpatialJoinImplAsync<>(duplicates, filter, null, null);
+    }
+
+    /**
+     * Creates and configures a new SpatialJoin object.
+     *
+     * @param duplicates Indicates whether the spatial join will suppress duplicates.
+     * @param <LEFT>     Type of object for left side of join
+     * @param <RIGHT>    Type of object for right side of join
+     * @return A configured SpatialJoin object. All spatial joins computed using it will use the configuration
+     * specified by the above arguments. False positives will be included in spatial join output.
+     */
+    public static <LEFT, RIGHT> SpatialJoinAsync<LEFT, RIGHT> newSpatialJoin(Duplicates duplicates)
+    {
+        return new SpatialJoinImplAsync<>(duplicates, null, null, null);
+    }
+
+    /**
+     * Returns an {@link java.util.Iterator} that will provide access to spatial join results.
```

Comment:
I think these comments about it returning a `java.util.Iterator` aren't quite right.